lib/tandem/tandemSimulator.js

```diff
@@ -119,6 +120,27 @@ exports.make = function(config){
         events.push(currBasal);
       }
     },
+    newDay: function(event) {
+      ensureTimestamp(event);
+      if((currBasal !== null) && (currBasal.deliveryType !== 'suspend')) {
+        // only insert the new day event if the basal rate is not suspended
+
+        if (currBasal.time !== event.time) {
+          if (!currBasal.isAssigned('duration')) {
+            currBasal.with_duration(Date.parse(event.time) - Date.parse(currBasal.time));
+          }
+          currBasal = currBasal.done();
+
+          event.set('type', 'basal');
+          event.previous = _.omit(currBasal, 'previous');
+          event.with_scheduleName(currBasal.scheduleName);
```

Comment:
Are you trying to rely on the fact that `currBasal.scheduleName` will almost always be `null` or `undefined` here? Schedule name on new day is just like schedule name on basals - can only be filled in on very final basal, as that's the only time we can guarantee it will be accurate (before we're reading and tracking a full settings history). If you're counting on the fallback to `null` or `undefined` here I'd prefer to use a different strategy - this is hard to read and understand.